File: angular_ui/src/app/student/model/student.ts
export interface Student {
     id: number,
     nic: string,
     firstName: string,
     lastName: string,
     dob: Date,
     regDate:Date,
     gender: string,
     currentStatus: string,
     telephones:string,
   //   telephones: [ {
   //      number: string
   //   }],

     addresses: {
      lane1: string,
      lane2: string,
      city: string
    }
}
// {
//     "id": 1,
//     "nic": "768668777",
//     "firstName": "lene",
//     "lastName": "lastlene",
//     "dob": "1994-10-10",
//     "regDate": "2020-10-10",
//     "gender": "Female",
//     "currentStatus": "Active",
//     "telephones": [{

// }],
//     "addresses": {
//         "id": 1,
//         "lane1": "lane 100",
//         "lane2": "lane 101",
//         "city": "kegalle"
//     }
// }

File: microservices/sms-commons/src/main/java/com/janani/sms/commons/model/student/Student.java
package com.janani.sms.commons.model.student;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDate;

@Data
@Entity
@Table(name = "student")
public class Student {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String nic;
    private String firstName;
    private String lastName;
    private LocalDate dob;
    private LocalDate regDate;
    private String gender;
    private String currentStatus;
    private String telephones;

//    @Enumerated(EnumType.STRING)
//    private Gender gender;
//
//    @Enumerated(EnumType.STRING)
//    private CurrentStatus currentStatus;

//    @OneToMany(mappedBy = "student", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
//    List<Telephone> telephones;

    @OneToOne(cascade = CascadeType.ALL)
    Address addresses;

}
